component/identity/src/test/java/org/exoplatform/services/organization/mock/InMemoryGroupHandler.java

```diff
@@ -175,7 +175,7 @@ public Group findGroupById(String groupId) {
 
   @Override
   public ListAccess<Group> findGroupChildren(Group parent, String keyword) {
-    List<Group> childGroups = groupChildsById.computeIfAbsent(parent.getId(), key -> Collections.emptyMap())
+    List<Group> childGroups = groupChildsById.computeIfAbsent(parent.getId(), key -> new HashMap<>())
```

Comment:
a `find...` method should never return a modifiable collection that **gives direct access** to cahce entries

```suggestion
    List<Group> childGroups = groupChildsById.computeIfAbsent(parent.getId(), key -> Collections.emptyMap())
```

Reply:
Here is my pb : 
In my test, I firstly do a call to 
findGroupByChildren : https://github.com/Meeds-io/gatein-portal/blob/2245c74981a4a302dcb436d6aefded12cde89e87/component/identity/src/test/java/org/exoplatform/services/organization/mock/InMemoryGroupHandler.java#L177
At this moment, the parent have no children, so the computeIfAbsent execute, and add an emptyMap (unmodifiable in groupChildsById.

Later, I try to add a children to the parent by calling saveGroup
So this is executed : 
https://github.com/Meeds-io/gatein-portal/blob/2245c74981a4a302dcb436d6aefded12cde89e87/component/identity/src/test/java/org/exoplatform/services/organization/mock/InMemoryGroupHandler.java#L123
`groupChildsById.computeIfAbsent(group.getParentId(), key -> new HashMap<String, Group>())
                   .put(group.getId(), ObjectUtils.clone(group));`
as the group.parentId exists, the computIfAbsent is not executed, and the put is launched on the unmodifiable map, which lead to UnsupporedOperationException

So I'm not sure to what I should do here.

Last but not least : if I call saveGroup, before the find, the list put in groupChildsById is a regular Hashmap, not a unmodifiable one.

What is your proposition on this point ?

Reply:
I see, thanks for the explanation. So you doesn't want to change the result of `find...` bu to change the `Map` stored in Cache.

No problem with that, just a little remark about title:
> When filling the cache in InMemoryGroupDaoHandler, do not return a non modifiable map
May be to change to
> fill cache of InMemoryGroupDaoHandler with a Modifiable Map

Reply:
Ok, I change the description